presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -271,29 +386,40 @@ private String toPredicate(String columnName, Domain domain, Type type, List<Typ
 
         // Add back all of the possible single values either as an equality or an IN predicate
         if (singleValues.size() == 1) {
-            disjuncts.add(toPredicate(columnName, "=", getOnlyElement(singleValues), type, accumulator));
+            disjuncts.add(toPredicate(columnName, "=", getOnlyElement(singleValues), columnHandle, accumulator, joinPushdownEnabled));
         }
         else if (singleValues.size() > 1) {
             for (Object value : singleValues) {
-                bindValue(value, type, accumulator);
+                bindValue(value, columnHandle, accumulator);
             }
             String values = Joiner.on(",").join(nCopies(singleValues.size(), "?"));
-            disjuncts.add(quote(columnName) + " IN (" + values + ")");
+            disjuncts.add(getColumnIdentifier(columnName, columnHandle, joinPushdownEnabled) + " IN (" + values + ")");
         }
 
         // Add nullability disjuncts
         checkState(!disjuncts.isEmpty());
         if (domain.isNullAllowed()) {
-            disjuncts.add(quote(columnName) + " IS NULL");
+            disjuncts.add(getColumnIdentifier(columnName, columnHandle, joinPushdownEnabled) + " IS NULL");
         }
 
         return "(" + Joiner.on(" OR ").join(disjuncts) + ")";
     }
 
-    private String toPredicate(String columnName, String operator, Object value, Type type, List<TypeAndValue> accumulator)
+    private String getColumnIdentifier(String columnName, JdbcColumnHandle columnHandle, boolean joinPushdownEnabled)
+    {
+        Optional<String> tableAlias = columnHandle.getTableAlias();
```

Comment:
Please remove the `joinPushdownEnabled` parameter. If a tableAlias is set, let's use it